### DISCUSSION TITLE
Is there any way to vacuum the db without having to stop pocketbase?

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#5387] [Q&A] [answered]
I'm using pocketbase in all my projects and it has proven to be very efficient and easy to use.

I recently started using it in a project that generates a slightly larger amount of data daily, and I'm saving it in pocketbase. 100k new records are created daily, they are small json data from logs from another service.

I initially thought it was normal for the data.db file to take up a few GB of space. However, I tested it on a copy of the DB by running the vacuum command, and the file got 10x smaller.

So I wondered if there is any way to "schedule" the execution of this command in the DB (maybe once a week).

I don't have much knowledge of Go to use pocketbase as a framework.
I use pocketbase via docker as it is provided.

Thanks for the great project.

**Comment by ganigeorgiev** [comment, marked as answer]
[`VACUUM`](https://www.sqlite.org/lang_vacuum.html) could be a very slow operation, especially for large datasets, so I'm not sure if it would be a good idea to run it unattended (or frequently), but if you really want to you can [schedule a cronjob](https://pocketbase.io/docs/js-jobs-scheduling/) that will run the raw query for you:

```js
// pb_hooks/main.pb.js

// every sunday at 00:00
cronAdd("vacuum", "0 0 * * 0", () => {
  // we explicitly use the nonconcurrentDB connection to minimize
  // the SQLITE_BUSY errors and avoid unnecessary triggering the fail/retry pooling
  $app.dao().nonconcurrentDB().newQuery("VACUUM").execute()
})
```
_(when deploying with Docker you need to copy or mount your `pb_hooks` folder next to the binary or specify explicitly the `--hooksDir` directory)_

**Reply by erickythierry** [reply]
Thank you very much for the answer. This specific DB is just logs for metrics in case of a problem in the main system, so it is not very important data. But thanks💜 for the warning, I will avoid doing this in instances with important data.

**Comment by abhranil26** [comment]
Fyi, after the [changes of v23](https://pocketbase.io/v023upgrade/jsvm/)
the new cron should be 
`// VACUUM main database - runs every Sunday at midnight
  cronAdd("vacuum", "0 0 * * 0", () => {
      $app.vacuum()
      console.log("VACUUM completed on [data.db](http://data.db/)")
  })`
